File: backend/app/services/automation_service.py
import os

import requests


def trigger_n8n(project) -> str:
    webhook_url = os.getenv("N8N_WEBHOOK_URL")
    if not webhook_url:
        return "Skipped: N8N_WEBHOOK_URL is not configured"

    payload = {
        "client_name": project.client_name,
        "project_name": project.project_type,
        "proposal_file": project.proposal_path,
        "ppt_file": project.ppt_path,
        "status": "completed",
        "user_email": project.user_email,
    }

    response = requests.post(webhook_url, json=payload, timeout=15)
    if response.status_code >= 400:
        return f"Failed with status {response.status_code}"

    return "Triggered successfully"

File: backend/src/services/automationService.js
export async function triggerN8n(client) {
  if (!process.env.N8N_WEBHOOK_URL) {
    return "Demo complete — add N8N_WEBHOOK_URL to send a real webhook";
  }

  const response = await fetch(process.env.N8N_WEBHOOK_URL, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({
      clientName: client.name,
      clientEmail: client.email,
      industry: client.industry,
      insights: client.insights,
      nextAction: client.insights?.nextAction,
    }),
    signal: AbortSignal.timeout(15000),
  });

  if (!response.ok) {
    throw new Error(`n8n returned status ${response.status}`);
  }

  return "Sent successfully";
}
